### ISSUE TITLE
dms_replication_task modifications to replication_task_settings requires recreation

### ISSUE THREAD

**Issue by fromonesrc** [hashicorp/terraform#12206]
### Terraform Version
0.8.7

### Affected Resource(s)
`dms_replication_task`

### Terraform Configuration Files
```
resource "aws_dms_replication_task" "task" {
  migration_type            = "${var.migration_type}"
  replication_instance_arn  = "${var.replication_instance_arn}"
  replication_task_id       = "${var.database_name}"
  replication_task_settings = "${trimspace(file("settings/settings.json"))}"
  source_endpoint_arn       = "${aws_dms_endpoint.source.endpoint_arn}"
  table_mappings            = "${var.table_mappings}"
  target_endpoint_arn       = "${var.warehouse_endpoint_arn}"
}
```

### Expected Behavior
When modifying the DMS task settings in settings.json, it should update the task settings.

### Actual Behavior
`* aws_dms_replication_task.task: ResourceNotReady: exceeded 10 wait attempts`

How I've been fixing this is by dropping the tasks from terraform, applying to delete them, then adding them back and applying again.

The `trimspace()` is also necessary or else terraform thinks there's a trailing new line on every run.

### Steps to Reproduce
1. Create a dms task with replication_task_settings in tf
1. Apply tf
1. Modify settings in tf
1. Apply tf

**Comment by atrepca** [comment]
I get `ResourceNotReady: exceeded 10 wait attempts` when the task is not in the 'Ready' status. Had 2 tasks in 'Error' and 'Stopped' for which the modification timed out, but the next `terraform apply` had no changes.

**Comment by atrepca** [comment]
@fromonesrc, the `trimspace` issue was fixed in hashicorp/terraform#12378.
